hazelcast-jet-core/src/test/java/com/hazelcast/jet/config/ResourceConfigTest.java

```diff
@@ -0,0 +1,327 @@
+        config.addJarsInZip(file);
+        ResourceConfig resourceConfig = config.getResourceConfigs().iterator().next();
+
+        assertEquals("zipFile", resourceConfig.getId());
+        assertEquals(ResourceType.JARS_IN_ZIP, resourceConfig.getResourceType());
+        assertEquals(file.toURI().toURL(), resourceConfig.getUrl());
+    }
+
+    @Test
+    public void testAttachFile_with_Url() throws Exception {
+        JobConfig config = new JobConfig();
+        String urlString = "file://path/to/resourceFile";
+        config.attachFile(new URL(urlString));
+        ResourceConfig resourceConfig = config.getResourceConfigs().iterator().next();
+
+        assertEquals("resourceFile", resourceConfig.getId());
+        assertEquals(ResourceType.FILE, resourceConfig.getResourceType());
+        assertEquals(urlString, resourceConfig.getUrl().toString());
+    }
+
+    @Test
+    public void testAttachFile_with_MalformedUrl_throws_Exception() throws Exception {
+        expectedException.expect(MalformedURLException.class);
+        JobConfig config = new JobConfig();
+        String urlString = "filezzz://path/to/file";
+        config.attachFile(new URL(urlString));
+    }
+
+    @Test
+    public void testAttachFile_with_Url_and_ResourceName() throws Exception {
+        JobConfig config = new JobConfig();
+        String resourceName = "resourceFileName";
+        String urlString = "file://path/to/resourceFile";
+        config.attachFile(new URL(urlString), resourceName);
+        ResourceConfig resourceConfig = config.getResourceConfigs().iterator().next();
+
+        assertEquals(resourceName, resourceConfig.getId());
+        assertEquals(ResourceType.FILE, resourceConfig.getResourceType());
+        assertEquals(urlString, resourceConfig.getUrl().toString());
+    }
+
+    @Test
+    public void testAttachFile_with_Path() throws Exception {
+        JobConfig config = new JobConfig();
+        String path = "/path/to/resource";
+        config.attachFile(path);
+        ResourceConfig resourceConfig = config.getResourceConfigs().iterator().next();
+
+        assertEquals("resource", resourceConfig.getId());
+        assertEquals(ResourceType.FILE, resourceConfig.getResourceType());
+        assertEquals(new File(path).toURI().toURL(), resourceConfig.getUrl());
+    }
+
+    @Test
+    public void testAttachFile_with_Path_and_ResourceName() throws Exception {
+        JobConfig config = new JobConfig();
+        String resourceName = "resourceFileName";
+        String path = "/path/to/jarfile";
+        config.attachFile(path, resourceName);
+        ResourceConfig resourceConfig = config.getResourceConfigs().iterator().next();
+
+        assertEquals(resourceName, resourceConfig.getId());
+        assertEquals(ResourceType.FILE, resourceConfig.getResourceType());
+        assertEquals(new File(path).toURI().toURL(), resourceConfig.getUrl());
+    }
+
+    @Test
+    public void testAttachFile_with_File() throws Exception {
```

Comment:
This test is exactly the same as `JobConfigTest.when_addResourceWithFile_thenReturnsResourceConfig`. Others probably too.